src/main/java/Duke.java

```diff
@@ -63,53 +70,176 @@ public static void main(String[] args) {
+        String[] parameters = new String[2];
+        getParameters(parameters, input, taskType);
+        switch (taskType) {
+        case TODO:
+            getParameters(parameters, input, TODO);
+            list[listIndexTracker] = new Todo(parameters[DESCRIPTION]);
+            break;
+        case DEADLINE:
+            getParameters(parameters, input, DEADLINE);
+            list[listIndexTracker] = new Deadline(parameters[DESCRIPTION], parameters[DATETIME]);
+            break;
+        case EVENT:
+            getParameters(parameters, input, EVENT);
+            list[listIndexTracker] = new Event(parameters[DESCRIPTION], parameters[DATETIME]);
+            break;
+        }
+
         listIndexTracker++;
+
     }
 
     /**
-     * This function prints the individual elements in list
+     * This function prints the message when task is added to list.
      */
-    public static void printList() {
+    private static void printAddedMessage() {
+        System.out.println(TAB + "Got it. I've added this task: ");
+        System.out.println(TAB + TAB + list[listIndexTracker - 1]);
+        System.out.println(TAB + String.format("Now you have %d tasks in the list.", listIndexTracker));
+    }
+
+    /**
+     * This function prints the individual elements in list.
+     */
+    private static void printList() {
         for (int i = 0; i < listIndexTracker; i++) {
-            System.out.format(TAB + "% 3d. [%s]", i + 1, list[i].getStatusIcon());
-            System.out.println(list[i].getDescription());
+            System.out.format(TAB + "% 3d.", i + 1);
+            System.out.println(list[i]);
+        }
+    }
+
+    /**
+     * This function sets done to true, stopping while loop and signifying the end of program.
+     */
+    private static void setDone() {
+        done = true;
+    }
+
+    /**
+     * This function check if done statement and index in done statement is valid.
+     * @param input input given by the user.
+     * @return returns the validity of the done input.
+     */
+    private static boolean isDoneInput(String input) {
+        boolean validDoneStatement = Pattern.matches("^done \\d+$", input.toLowerCase());
+        if (!validDoneStatement) {
+            return false;
+        }
+        int numberOnly = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+        return numberOnly <= listIndexTracker && numberOnly < 100;
+    }
+
+    /**
+     * This function retrieves the index given in done statement.
+     * @param input input given by the user.
+     * @return returns the index given in done statement
+     */
+    private static int getDoneIndex(String input) {
+        String[] parts = input.split(" ");
+        return Integer.parseInt(parts[1]) - 1;
+    }
+
+    /**
+     * This function sets error as the given boolean argument.
+     * @param bool boolean to set error as.
+     */
+    private static void setError(boolean bool) {
```

Comment:
The boolean bool could have been named better. It is hard to understand what it refers to. Preferably a name with prefix is, has or was.